federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -604,13 +604,7 @@ func (ic *IngressController) getMasterCluster() (master *federationapi.Cluster,
 */
 func (ic *IngressController) updateClusterIngressUIDToMasters(cluster *federationapi.Cluster, fallbackUID string) (string, error) {
 	masterCluster, masterUID, err := ic.getMasterCluster()
-	clusterObj, clusterErr := api.Scheme.DeepCopy(cluster) // Make a clone so that we don't clobber our input param
-	cluster, ok := clusterObj.(*federationapi.Cluster)
-	if clusterErr != nil || !ok {
-		glog.Errorf("Internal error: Failed clone cluster resource while attempting to add master ingress UID annotation (%q = %q) from master cluster %q to cluster %q, will try again later: %v", uidAnnotationKey, masterUID, masterCluster.Name, cluster.Name, clusterErr)
-		return "", clusterErr
-	}
-
+	cluster = cluster.DeepCopy() // Make a clone so that we don't clobber our input param
```

Comment:
It seems we lose a lot of error detection and logging here, and more specifically will panic and crash rather than retry.  Or am I misunderstanding something here?

Reply:
DeepCopy cannot fail (anymore). There is no need for the theoretical corner cases we had to take care of before.

Reply:
Everything we checked before with code (the type of `clusterObj` and the ability to deepcopy it) is now guaranteed by the type-system, i.e. by the Go compiler. `cluster.DeepCopy()` gives you either a deep copy (always!) or nil. The corner cases of before are gone.

Reply:
Awesome.  That's good to know.  I agree.